rust/operator-binary/src/airflow_controller.rs

```diff
@@ -602,68 +634,15 @@ async fn build_executor_template(
     Ok(())
 }
 
-/// The server-role service is the primary endpoint that should be used by clients that do not perform internal load balancing,
-/// including targets outside the cluster.
-fn build_role_service(
-    airflow: &v1alpha1::AirflowCluster,
-    resolved_product_image: &ResolvedProductImage,
-    role_name: &str,
-    port: u16,
-) -> Result<Service> {
-    let role_svc_name = format!("{}-{}", airflow.name_any(), role_name);
-    let ports = role_ports(port);
-
-    let metadata = ObjectMetaBuilder::new()
-        .name_and_namespace(airflow)
-        .name(&role_svc_name)
-        .ownerreference_from_resource(airflow, None, Some(true))
-        .context(ObjectMissingMetadataForOwnerRefSnafu)?
-        .with_recommended_labels(build_recommended_labels(
-            airflow,
-            AIRFLOW_CONTROLLER_NAME,
-            &resolved_product_image.app_version_label,
-            role_name,
-            "global",
-        ))
-        .context(ObjectMetaSnafu)?
-        .build();
-
-    let service_selector_labels =
-        Labels::role_selector(airflow, APP_NAME, role_name).context(BuildLabelSnafu)?;
-
-    let service_spec = ServiceSpec {
-        type_: Some(
-            airflow
-                .spec
-                .cluster_config
-                .listener_class
-                .k8s_service_type(),
-        ),
-        ports: Some(ports),
-        selector: Some(service_selector_labels.into()),
-        ..ServiceSpec::default()
-    };
-
-    Ok(Service {
-        metadata,
-        spec: Some(service_spec),
-        status: None,
-    })
-}
-
 fn role_ports(port: u16) -> Vec<ServicePort> {
```

Comment:
I don't love the kinda weird roundabout way this is bounced around, but not new so oh well.